File: services/basket/src/main/java/org/cryptimeleon/incentive/basket/model/BasketItem.java
package org.cryptimeleon.incentive.basket.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Utility dataclass for representing Items in a Basket.
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class BasketItem {
    Item item;
    int count;
}

File: android/app/src/main/java/org/cryptimeleon/incentive/app/domain/model/BasketItem.kt
package org.cryptimeleon.incentive.app.domain.model

import org.cryptimeleon.incentive.promotion.model.BasketItem


data class BasketItem(
    val itemId: String,
    val title: String,
    val price: Int,
    val count: Int,
) {
    fun toPromotionBasketItem(): BasketItem {
        return BasketItem(itemId, title, price, count)
    }
}
